src/lib/react-query/projects.ts

```diff
@@ -394,6 +395,44 @@ export function updateProjectSettingsMutationOptions({
 	>
 }
 
+export function changeMemberRoleMutationOptions({
+	projectApi,
+	projectId,
+	queryClient,
+}: {
+	projectApi: MapeoProjectApi
+	projectId: string
+	queryClient: QueryClient
+}) {
+	return {
+		...baseMutationOptions(),
+		mutationFn: async ({
+			deviceId,
+			roleId,
+		}: {
+			deviceId: string
+			roleId: MemberApi.RoleIdForNewInvite
+		}) => {
+			return projectApi.$member.assignRole(deviceId, roleId)
+		},
+		onSuccess: (_data, { deviceId }) => {
+			queryClient.invalidateQueries({
+				queryKey: getMembersQueryKey({ projectId }),
+			})
+			queryClient.invalidateQueries({
+				queryKey: getMemberByIdQueryKey({ projectId, deviceId }),
+			})
```

Comment:
no need for the device-specific invalidation as the members invalidation just before it will cascade to include it
```suggestion
		onSuccess: () => {
			queryClient.invalidateQueries({
				queryKey: getMembersQueryKey({ projectId }),
			})
```